lib/keyConnections.js

```diff
@@ -72,28 +73,51 @@ const connectionPointTo = (element) => {
  */
 export const getKeyConnections = () => {
   // A list of all the key pointers in the DOM
-  const renderedKeyPointers = document.getElementsByClassName(keyInfoConnectFromClass)
+  const renderedKeyPointers = document.getElementsByClassName(
+    keyInfoConnectFromClass
+  );
 
-  var connections = []
+  var connections = [];
+  var targetKeys = [];
   for (let keyPointer of renderedKeyPointers) {
-    const sourceCoords = connectionPointFrom(keyPointer)
+    const sourceCoords = connectionPointFrom(keyPointer);
 
     // Convert the key pointer class names to a list of bare key pointers,
     // e.g. key-info-connect-from-l-f-1-1 => l-f-1-1
-    const targetKeyIds = keyPointer
-      .className
-      .split(' ')
-      .filter(cls => cls.startsWith(keyInfoConnectFromClassPrefix))
-      .map(cls => cls.slice(keyInfoConnectFromClassPrefix.length))
+    const targetKeyIds = keyPointer.className
+      .split(" ")
+      .filter((cls) => cls.startsWith(keyInfoConnectFromClassPrefix))
+      .map((cls) => cls.slice(keyInfoConnectFromClassPrefix.length));
 
-    targetKeyIds.forEach(targetKeyId => {
-      const targetCoords = connectionPointTo(document.getElementById(targetKeyId))
+    targetKeyIds.forEach((targetKeyId) => {
+      targetKeys.push(targetKeyId);
+      const targetKey = document.getElementById(targetKeyId);
+      const targetCoords = connectionPointTo(targetKey);
       log.debug(
         `Found key connection from source at ${sourceCoords.x},${sourceCoords.y}`,
         `to target key with ID ${targetKeyId} at ${targetCoords.x},${targetCoords.y}`
-      )
-      connections.push([sourceCoords, targetCoords])
-    })
+      );
+      connections.push([sourceCoords, targetCoords]);
+    });
   }
-  return connections;
-}
\ No newline at end of file
+  return {
+    connections,
+    targetKeyIds: targetKeys,
+  };
+};
+
+export const useKeyConnections = (effectProps = []) => {
+  const [connections, setConnections] = useState();
+  const [targetKeyIds, setTargetKeyIds] = useState();
+
+  useEffect(() => {
+    const data = getKeyConnections();
+    setConnections(data.connections);
+    setTargetKeyIds(data.targetKeyIds);
+  }, effectProps);
```

Comment:
the `effectProps` here contains the window size. When that changes, we run this function.